cypress/integration/catalog/collections.js

```diff
@@ -216,7 +223,126 @@ filterTests({ definedTags: ["all"] }, () => {
+          .find(BUTTON_SELECTORS.checkbox)
+          .click()
+          .get(collectionRow(secondCollection.id))
+          .find(BUTTON_SELECTORS.checkbox)
+          .click()
+          .get(BUTTON_SELECTORS.deleteIcon)
+          .click()
+          .addAliasToGraphRequest("CollectionBulkDelete")
+          .get(BUTTON_SELECTORS.submit)
+          .click()
+          .waitForRequestAndCheckIfNoErrors("@CollectionBulkDelete");
+
+        getCollection({ collectionId: firstCollection.id, auth: "auth" })
+          .its("collection")
+          .should("be.null");
+        getCollection({ collectionId: secondCollection.id, auth: "auth" })
+          .its("collection")
+          .should("be.null");
+      });
+    });
+
+    it("assign product to collection. TC: SALEOR_0307", () => {
+      cy.log("przed tworzeniem imion");
+      const collectionName = `Assign-${startsWith}${faker.datatype.number()}`;
+      const productName = `Product-To-Assign-${startsWith}${faker.datatype.number()}`;
+
+      let collection;
+      let productToAssign;
+      cy.log("before request");
+
+      createCollectionRequest(collectionName).then(collectionResp => {
+        collection = collectionResp;
+
+        productsUtils
+          .createProductInChannel({
+            name: productName,
+            channelId: defaultChannel.id,
+            productTypeId: productType.id,
+            attributeId: attribute.id,
+            categoryId: category.id,
+            visibleInListings: false
+          })
+          .then(({ product: productResp }) => {
+            productToAssign = productResp;
+
+            cy.visit(collectionDetailsUrl(collection.id));
+            assignProductsToCollection(productToAssign.name);
+
+            getCollection({ collectionId: collection.id, auth: "auth" })
+              .its("collection.products.edges")
+              .should("have.length", 1)
+              .then(productArray => {
+                expect(productArray[0].node.id).to.equal(productToAssign.id);
+              });
+          });
+      });
+    });
+
+    it("remove product from collection. TC: SALEOR_0308", () => {
+      const collectionName = `Remove-With-Assigned-Product-${startsWith}${faker.datatype.number()}`;
+      const productName = `Product-To-Assign-${startsWith}${faker.datatype.number()}`;
+      let collection;
+      let productToAssign;
+
+      createCollectionRequest(collectionName).then(collectionResp => {
+        collection = collectionResp;
+
+        productsUtils
+          .createProductInChannel({
+            name: productName,
+            channelId: defaultChannel.id,
+            productTypeId: productType.id,
+            attributeId: attribute.id,
+            categoryId: category.id,
+            visibleInListings: false
+          })
+          .then(({ product: productResp }) => {
+            productToAssign = productResp;
+
+            cy.visit(collectionDetailsUrl(collection.id));
+
+            addProductToCollection({
```

Comment:
Maybe first add product to collection, and then visit collection test. It might by flaky